File: InterviewHub/src/main/java/com/InterviewHub/feature/auth/AuthService.java
package com.InterviewHub.feature.auth;

import com.InterviewHub.dto.AuthRequest;
import com.InterviewHub.dto.AuthResponse;
import com.InterviewHub.dto.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(AuthRequest request);
    com.InterviewHub.feature.user.dto.UserResponse getAccount();
}

File: FrontEnd/my-react-app/src/services/authService.js
import axiosClient from './axiosClient';

export const authService = {
  register: (data) => {
    return axiosClient.post('/auth/register', data);
  },
  login: (data) => {
    return axiosClient.post('/auth/login', data);
  },
  getAccount: () => {
    return axiosClient.get('/auth/account');
  }
};
